sources/api/netdog/src/networkd_status.rs

```diff
@@ -0,0 +1,304 @@
+//! The networkd_status module contains definitions and functions for tracking network status from networkd
+//!
+use crate::interface_id::InterfaceName;
+#[cfg(net_backend = "systemd-networkd")]
+use crate::NETWORKCTL;
+use serde::{Deserialize, Deserializer};
+#[cfg(net_backend = "systemd-networkd")]
+use snafu::{ensure, ResultExt};
+#[cfg(net_backend = "systemd-networkd")]
+use std::cmp::Ordering;
+use std::convert::TryInto;
+use std::net::IpAddr;
+#[cfg(net_backend = "systemd-networkd")]
+use std::process::Command;
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
```

Comment:
I left this in a strange spot. It feels unsafe to be less strict with matching this in case things change, so I wanted to match specifically. But it's probably fine to trim down to what we need and not worry about extra fields. I'm torn between only having the minimum for what we need today and spending a lot of effort adding in everything just in case. But I do see some possible needs of some of the various fields I've called out but just not in this PR. So this line was sort of here to force this particular thread of conversations. Do we want to be minimal and know there is a bunch of things we might need to add later or do we want to attempt to be exhaustive?